transport/peerlist.go

```diff
@@ -0,0 +1,15 @@
+package transport
+
+import "context"
+
+//go:generate mockgen -destination=transporttest/peerlist.go -package=transporttest go.uber.org/yarpc/transport PeerList
+
+// PeerList is a collection of Peers.  Outbounds request peers from the ChoosePeer to determine where to send requests
+type PeerList interface {
+	PeerSubscriber
```

Comment:
I did this mostly as a way of hiding start/stop/choose information from Peers.  A PeerList wants to get information from a peer when it connects/pending count changes/etc but I didn't want to combine the Subscription interface with the start/stop/choose interface.

There is a good argument to be made that PeerList and PeerSubscriber should be two separate interfaces which I am fine with, lists will just have to implement both (or have separate Lists and Subscribers).